server/repository/src/db_diesel/changelog/changelog.rs

```diff
@@ -205,16 +378,80 @@ fn create_filtered_query(earliest: u64, filter: Option<ChangelogFilter>) -> Boxe
             record_id,
             is_sync_update,
             action,
+            source_site_id,
         } = f;
 
         apply_equal_filter!(query, table_name, changelog_deduped::table_name);
         apply_equal_filter!(query, name_id, name_link::name_id);
         apply_equal_filter!(query, store_id, changelog_deduped::store_id);
         apply_equal_filter!(query, record_id, changelog_deduped::record_id);
-        apply_equal_filter!(query, is_sync_update, changelog_deduped::is_sync_update);
         apply_equal_filter!(query, action, changelog_deduped::row_action);
+        apply_equal_filter!(query, is_sync_update, changelog_deduped::is_sync_update);
+        apply_equal_filter!(query, source_site_id, changelog_deduped::source_site_id);
+    }
+
+    query
+}
+
+/// This looks up associated records to decide if change log should be sent to the site or not
+/// Update this method when adding new sync styles to the system
+fn create_filtered_outgoing_sync_query(
+    earliest: u64,
+    sync_site_id: i32,
+    is_initialized: bool,
+) -> BoxedChangelogQuery {
+    let mut query = changelog_deduped::table
+        .left_join(name_link::table)
+        .filter(changelog_deduped::cursor.ge(earliest.try_into().unwrap_or(0)))
+        .into_boxed();
+
+    // If we are initialising, we want to send all the records for the site, even ones that originally came from the site
+    // The rest of the time we want to exclude any records that were created by the site
+
+    if is_initialized {
+        query = query
+            .filter(
+                changelog_deduped::source_site_id
+                    .ne(Some(sync_site_id.clone()))
+                    .or(changelog_deduped::source_site_id.is_null()),
+            )
+            .filter(changelog_deduped::is_sync_update.eq(false))
     }
 
+    // Loop through all the Sync tables and add them to the query if they have the right sync style
+
+    // Central Records
+
+    let central_sync_table_names: Vec<ChangelogTableName> = ChangelogTableName::iter()
```

Comment:
I do too, I had a crack at it but seems like you'd need to implement it on the time rather than an enum value somehow. Not sure how I can do that. Feel free to suggest a way! I see strum does it by implementing a trait maybe that's some how required, or I'm missing something basic...